pkg/cli/admin/upgrade/status/alerts.go

```diff
@@ -0,0 +1,73 @@
+package status
+
+import (
+	"time"
+)
+
+type AlertLabels struct {
+	AlertName string `json:"alertname,omitempty"`
+	Name      string `json:"name,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+	Reason    string `json:"reason,omitempty"`
+	Severity  string `json:"severity,omitempty"`
+}
+
+type AlertAnotations struct {
+	Description string `json:"description,omitempty"`
+	Summary     string `json:"summary,omitempty"`
+}
+
+type Alert struct {
+	Labels      AlertLabels     `json:"labels,omitempty"`
+	Annotations AlertAnotations `json:"annotations,omitempty"`
+	State       string          `json:"state,omitempty"`
+	Value       string          `json:"value,omitempty"`
+	ActiveAt    time.Time       `json:"activeAt,omitempty"`
+	Level       string          `json:"partialResponseStrategy,omitempty"`
+}
+
+type AlertData struct {
+	Status string `json:"status"`
+	Data   Data   `json:"data"`
+}
+
+type Data struct {
+	Alerts []Alert `json:"alerts"`
+}
+
+func parseAlertDataToInsights(alertData AlertData, startedAt time.Time) []updateInsight {
+	var alerts []Alert = alertData.Data.Alerts
+	var updateInsights []updateInsight
+
+	for _, alert := range alerts {
+		if startedAt.After(alert.ActiveAt) {
+			continue
```

Comment:
No worries if you want to push back, but [OTA-1157](https://issues.redhat.com/browse/OTA-1157)'s current asks include:

> Alerts that started firing before the upgrade but are present on an allowlist (hardcoded for now) of alerts relevant for update

so we can pass along `PodDisruptionBudgetLimit` and `PodDisruptionBudgetAtLimit` and such, even if they'd started firing before this update was initiated.